### PR TITLE
[cmake] Fix ninja (fixes: #15008)

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

remove duplicate output target for addons/xbmc.json/addon.xml

## Motivation and Context

see #15008

## How Has This Been Tested?

only quickly on ubuntu. I'm not 100% certain this does not have an side effects...

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: cmake/scripts/common/Macros.cmake [hunk 1]
```diff
@@ -776,7 +776,11 @@ endmacro()
 # add-on xml's
 # find all folders containing addon.xml.in and used to define
 # ADDON_XML_OUTPUTS, ADDON_XML_DEPENDS and ADDON_INSTALL_DATA
-macro(find_addon_xml_in_files)
+macro(find_addon_xml_in_files filter)
```
Comment (Rechi) [inline cmake/scripts/common/Macros.cmake]: Is there any reason to force passing at least one filter?

File: cmake/scripts/common/Macros.cmake [hunk 2]
```diff
@@ -785,7 +789,9 @@ macro(find_addon_xml_in_files)
     string(REPLACE "${CORE_SOURCE_DIR}/" "" xml_name ${xml_name})
 
     list(APPEND ADDON_XML_DEPENDS "${CORE_SOURCE_DIR}/${xml_name}/addon.xml.in")
-    list(APPEND ADDON_XML_OUTPUTS "${CMAKE_BINARY_DIR}/${xml_name}/addon.xml")
+    if(NOT ${xml_name} MATCHES ${filter})
```
Comment (Rechi) [inline cmake/scripts/common/Macros.cmake]: As filter could be a list MATCHES is wrong.
You should ensure that filter either is always a string or always a list.

**Comment by wsnipex** [comment]
updated

File: cmake/scripts/common/Macros.cmake [hunk 3]
```diff
@@ -777,7 +777,15 @@ endmacro()
...
 # ADDON_XML_OUTPUTS, ADDON_XML_DEPENDS and ADDON_INSTALL_DATA
 macro(find_addon_xml_in_files)
+  set(filter ${ARGN})
+
   file(GLOB ADDON_XML_IN_FILE ${CMAKE_SOURCE_DIR}/addons/*/addon.xml.in)
+  if(filter)
+    if(VERBOSE)
+      message(STATUS "find_addon_xml_in_files: filtering ${filter}")
+    endif()
+    list(FILTER ADDON_XML_IN_FILE EXCLUDE REGEX "${filter}")
```
Comment (Rechi) [inline cmake/scripts/common/Macros.cmake]: isn't correct as the filter **list** could have more then one item

Comment (wsnipex) [reply cmake/scripts/common/Macros.cmake]: I made it a single regex now, no need for a list and another loop

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by MilhouseVH** [comment]
Tested this with LibreELEC and it's all good with the addition of the fixup commit (xbmc.json/addon.xml isn't installed without the fixup commit). Thanks @wsnipex

**Comment by wsnipex** [comment]
jenkins build this please

**Review by Rechi — APPROVED** [review]

**Comment by candrews** [comment]
Can this please be considered for Kodi 18?

**Comment by wsnipex** [comment]
I'd like confirmation that it doesn't somehow break windows first

**Comment by candrews** [comment]
On that note, should Travis and/or Appveyor be configured to use ninja?

**Comment by DaveTBlake** [comment]
>I'd like confirmation that it doesn't somehow break windows first

Seems fine @wsnipex ran a Windows x64 build and JSON commands were working as expected.

**Comment by gentoo-junkie** [comment]
Hi, could you please merge into feature_python3 branch and I will test it there.

**Pull request merged**